DeadCrumbs/app/src/main/java/dead/crumbs/ui/DeadReckoningService.kt

```diff
@@ -0,0 +1,73 @@
+package dead.crumbs.ui
+
+import android.content.Context
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
+import android.util.Log
+import android.app.Service
+import android.content.Intent
+import android.os.Binder
+import android.os.IBinder
+
+class DeadReckoningService : Service(), SensorEventListener{
+
+    private lateinit var sensorManager: SensorManager
+    private var stepCounterSensor: Sensor? = null
+
+    override fun onCreate() {
+        super.onCreate()
+
+        sensorManager = getSystemService(Context.SENSOR_SERVICE) as SensorManager
+        stepCounterSensor = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_COUNTER)
+
+        sensorManager.registerListener(this, stepCounterSensor, SensorManager.SENSOR_DELAY_NORMAL)
+
+    }
+
+    //----------Binding--------------
+    // Binder given to clients
+    private val binder = LocalBinder()
+
+    /**
+     * Class used for the client Binder.  Because we know this service always
+     * runs in the same process as its clients, we don't need to deal with IPC.
+     */
+
+    inner class LocalBinder : Binder(){
+        // Return this instance of LocalService so clients can call public methods
+        fun getService(): DeadReckoningService = this@DeadReckoningService
+    }
+
+    override fun onBind(intent: Intent?): IBinder? {
+        return binder
+    }
+
+    private var stepCounterInitial : Int = 0
+    private var stepCounter : Int = 0
+
+    /* This is called whenever this class (SensorEventListener) detects a new sensor value
+     * from a sensor it is listening to (registerListener) */
+    override fun onSensorChanged(event: SensorEvent) {
+        if (event.sensor.type == Sensor.TYPE_STEP_COUNTER){
```

Comment:
Altså det er jo kun stepCounterSensoren der er sat som listener så lige nu er tjekket ikke nødvendigt.

Reply:
I believe this may just be good practice to include, I can't remember seeing an onSensorChanged without it. Could be wrong tho.